examples/jetstream/workqueue-stream/java/Main.java

```diff
@@ -0,0 +1,171 @@
+      // to create another consumer, it will fail.
+      ConsumerConfiguration c2 = ConsumerConfiguration.builder()
+        .durable("processor-2")
+        .ackPolicy(AckPolicy.Explicit)
+        .build();
+
+      try {
+        jsm.addOrUpdateConsumer("EVENTS", c2);
+      } catch (JetStreamApiException e) {
+        System.out.println("Failed to create an overlapping consumer");
+      }
+
+      // However if we delete the first consumer, this will clean up
+      // the first consumer since it was an ephemeral, which removes interest.
+      jsm.deleteConsumer("EVENTS", "processor-1");
+      System.out.println("Deleting the first consumer");
+
+      jsm.addOrUpdateConsumer("EVENTS", c2);
+      System.out.println("Succeeded to create a new consumer");
+      jsm.deleteConsumer("EVENTS", "processor-2");
+
+      // ### Multiple filtered consumers
+      // To create multiple consumers, we need to filter the subjects.
+      // For example, we can create a consumer that only receives
+      // subjects that start with `events.us` and another that only
+      // receives subjects that start with `events.eu`.
+      ConsumerConfiguration c3 = ConsumerConfiguration.builder()
+        .durable("processor-3")
+        .ackPolicy(AckPolicy.Explicit)
+        .filterSubject("events.us.>")
+        .build();
+
+      jsm.addOrUpdateConsumer("EVENTS", c3);
+
+      PullSubscribeOptions o3 = PullSubscribeOptions.bind("EVENTS", "processor-3");
+
+      JetStreamSubscription s3 = js.subscribe("", o3);
+
+      ConsumerConfiguration c4 = ConsumerConfiguration.builder()
+        .durable("processor-4")
+        .ackPolicy(AckPolicy.Explicit)
+        .filterSubject("events.eu.>")
+        .build();
+
+      jsm.addOrUpdateConsumer("EVENTS", c4);
+
+      PullSubscribeOptions o4 = PullSubscribeOptions.bind("EVENTS", "processor-4");
+
+      JetStreamSubscription s4 = js.subscribe("", o4);
+      System.out.println("Created two filtered consumers");
+
+      js.publish("events.us.page_loaded", null);
+      js.publish("events.eu.mouse_clicked", null);
+      js.publish("events.us.input_focused", null);
+      System.out.println("Published 3 more messages");
+
+      s3.pull(2);
+      for (int i = 0; i < 2; i++) {
+        Message msg = s3.nextMessage(Duration.ofSeconds(1));
+        msg.ack();
+        System.out.printf("Received event via 'processor-us': %s\n", msg.getSubject());
+      }
+
+      s4.pull(1);
+      Message msg = s4.nextMessage(Duration.ofSeconds(1));
+      msg.ack();
+      System.out.printf("Received event via 'processor-eu': %s\n", msg.getSubject());
+
+      s3.unsubscribe();
+      jsm.deleteConsumer("EVENTS", "processor-3");
+
+      s4.unsubscribe();
+      jsm.deleteConsumer("EVENTS", "processor-4");
+
+      // Sleep this thread a little so the dispatcher thread has time
+      // to receive all the messages before the program quits.
+      Thread.sleep(200);
+
```

Comment:
there is no dispatcher in pull, only in push async, can remove this sleep and comment

Reply:
Removed.